### PR TITLE
Allow noncompact CompactDecimal

### PR DESCRIPTION
**Pull request by eggrobin**
As discussed with @sffc; towards https://github.com/unicode-org/icu4x/issues/1267.

### CODE REVIEW

File: utils/fixed_decimal/src/compact.rs [hunk 1]
```diff
@@ -23,6 +25,16 @@ pub struct CompactDecimal {
     exponent: i16,
 }
 
+impl CompactDecimal {
+    pub fn significand(self) -> FixedDecimal {
```
Comment (sffc) [inline utils/fixed_decimal/src/compact.rs:29]:
```suggestion
    pub fn into_significand(self) -> FixedDecimal {
```

File: utils/fixed_decimal/src/compact.rs [hunk 2]
```diff
@@ -23,6 +25,16 @@ pub struct CompactDecimal {
...
+        self.significand
+    }
+
+    pub fn exponent(self) -> i16 {
```
Comment (sffc) [inline utils/fixed_decimal/src/compact.rs:33]:
```suggestion
    pub fn exponent(&self) -> i16 {
```

File: utils/fixed_decimal/src/compact.rs [hunk 3]
```diff
@@ -12,8 +12,10 @@ use crate::FixedDecimal;
 
 /// A struct containing a [`FixedDecimal`] significand together with an exponent, representing a
 /// number written in compact notation (such as 1.2M).
-/// This represents a _source number_ that uses compact decimal notation, as defined
+/// This represents a _source number_, as defined
 /// [in UTS #35](https://www.unicode.org/reports/tr35/tr35-numbers.html#Plural_rules_syntax).
+/// The value exponent=0 is used internally to represent a number in non-compact
```
Comment (sffc) [inline utils/fixed_decimal/src/compact.rs:17]:
```suggestion
/// The value exponent=0 represents a number in non-compact
```

**Review by sffc — CHANGES_REQUESTED** [review]

**Comment by sffc** [comment]
Also please add docs to the two new functions

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
Bumping CI

**Comment by sffc** [comment]
```
[cargo-make] ERROR - Error while running duckscript: Source: Unknown Line: 6 - Diplomat coverage dump need to be regenerated. Please run `cargo make diplomat-coverage` and commit.
```

File: utils/fixed_decimal/src/compact.rs [hunk 4]
```diff
@@ -92,10 +136,6 @@ fn test_compact_syntax_error() {
         pub expected_err: Option<Error>,
     }
     let cases = [
-        TestCase {
```
Comment (Manishearth) [inline utils/fixed_decimal/src/compact.rs:95]: nit: probably add some testcases that ensure it works now

**Review by Manishearth — APPROVED** [review]

File: utils/fixed_decimal/src/compact.rs (hunk 4 above)
Comment (eggrobin) [reply utils/fixed_decimal/src/compact.rs:95]: They are in the doc tests (and they even caught a hint bug); the `#[test]` only covers errors.

Comment (Manishearth) [reply utils/fixed_decimal/src/compact.rs:95]: ah neat

**Review by Manishearth — APPROVED** [review]

**Pull request merged**